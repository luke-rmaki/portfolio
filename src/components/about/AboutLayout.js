import styled from 'styled-components';

const AboutLayout = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;

  h1 {
    place-self: flex-start;
    padding: 100px 0 30px 50px;
  }

  aside {
    padding: 0 0 150px 0;
  }
`;

export default AboutLayout;
