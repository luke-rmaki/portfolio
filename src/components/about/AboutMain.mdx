---
name: about
title: about
edited: '31/12/2020'
links: https://github.com/luke-rmaki/, https://www.linkedin.com/in/luke-ruokaismaki-01b537159/, https://twitter.com/luke_rmaki, https://codepen.io/luke_rmaki
---

import AboutImage from './Image.js';

Hi, I'm **Luke**. I'm a Web Developer from Melbourne,
Australia.

<AboutImage />

I love working on the **front-end** and **back-end**. While I work primarily in **JavaScript** (I'm very excited about the future of **[Deno](https://deno.land/)**), I first learn to code with **Python**. I've also worked with **PHP** and
**Wordpress**, both by myself and on a team.

I am in the process of dabbling in **Rust** and am already pretty excited to try and build something with it.

## Some things I've worked on

// Under construction

---
