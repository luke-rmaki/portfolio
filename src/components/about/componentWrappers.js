import styled from 'styled-components';

const Wrapper = styled.div`
  grid-column: 1 / 13;
`;

const HeadingWrapper = styled(Wrapper)`
  h1 {
    margin: 150px 10px;
    @media screen and (min-width: 1500px) {
      margin: 200px 20px;
    }
  }
`;

const ImageWrapper = styled(Wrapper)`
  > div {
    transform: translateY(-50%);
  }
`;

const SocialMediaWrapper = styled(Wrapper)`
  text-align: left;
  margin-left: 10px;
`;

export { HeadingWrapper, ImageWrapper, SocialMediaWrapper };
