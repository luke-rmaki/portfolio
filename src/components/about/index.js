export { default as AboutLayout } from './AboutLayout';
export { default as SocialMedia } from './SocialMedia';
export { default as StyledContent } from './StyledContent';
