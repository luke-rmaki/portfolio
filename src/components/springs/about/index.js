export { default as ImageSpring } from './ImageSpring';
export { default as ContentSpring } from './ContentSpring';
export { default as SocialMediaSpring } from './SocialMediaSpring';
