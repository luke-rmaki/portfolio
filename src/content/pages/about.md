---
name: 'about'
links: https://github.com/luke-rmaki/, https://www.linkedin.com/in/luke-ruokaismaki-01b537159/
---

Some stuff about me

## I'm also on...
