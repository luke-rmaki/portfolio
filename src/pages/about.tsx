import React from 'react';
import styled from 'styled-components';
import { useStaticQuery, graphql } from 'gatsby';
import { MDXRenderer } from 'gatsby-plugin-mdx';

// Component imports
import { SocialMedia } from '../components/contact/SocialMedia';
import { Layout } from '../components/global/Layout';
import { Menu } from '../components/global/Menu';
import { MainHeading } from '../components';
import { Projects } from '../components/projects';

const About = (props) => {
  const { path }: { path: string } = props;

  const content = useStaticQuery(graphql`
    query AboutContent {
      mdx(frontmatter: { title: { eq: "about" } }) {
        body
      }
    }
  `);
  return (
    <Layout title="About">
      <Menu path={path} />
      <StyledAbout>
        <MainHeading>About</MainHeading>
        <article>
          <MDXRenderer>{content.mdx.body}</MDXRenderer>
        </article>
        <Projects />
        <hr />
        <SocialMedia />
      </StyledAbout>
    </Layout>
  );
};

export default About;

const StyledAbout = styled.main`
  article {
    padding: 0;
    display: grid;
    grid-template-columns: 1fr 2fr;

    p,
    span,
    blockquote {
      line-height: 40px;
    }

    [data-img] {
      grid-column: 1 / 3;
    }

    @media (min-width: 1000px) {
      [data-img] {
        grid-column: 2 / 3;
        justify-self: end;
        align-self: start;
        grid-row: 2 / 5;
      }
    }

    blockquote {
      background: var(--purple);

      a {
        color: var(--accent);
      }

      &:nth-child(2n) {
        background: var(--black);

        a {
          color: var(--white);
        }
      }
    }

    p,
    h2,
    blockquote {
      padding: 20px;
    }

    @media (max-width: 400px) {
      p,
      h2,
      blockquote {
        padding: 10px;
      }
    }

    p,
    blockquote {
      font-family: var(--font);
      font-size: var(--paragraph-size);
      color: var(--white);
      text-indent: 10px;
      margin-bottom: 20px;
      grid-column: 1 / 3;

      strong {
        font-weight: bold;
        font-style: italic;
        color: var(--accent);
      }

      mark {
        background: var(--accent);
        color: var(--black);
      }
    }

    a {
      color: var(--purple);
      text-decoration: none;
    }

    h2 {
      font-family: var(--font);
      font-weight: normal;
      font-style: italic;
      font-size: var(--subheading-size);
      color: var(--purple);
    }
  }

  hr {
    margin: 100px 0;
  }

  @media (min-width: 1000px) {
    article {
      grid-template-columns: 50% 50%;

      p,
      blockquote {
        grid-column: 1 / 2;
      }
    }
  }
`;
