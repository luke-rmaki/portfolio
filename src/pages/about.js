import React from 'react';

// Component imports
import Menu from '../components/nav/Menu';
import { AboutImage, AboutContent, SocialMedia } from '../components/content';
import { Layout, Spacer, Flex } from '../components/layout';
import { MainHeading, Subheading } from '../components/typography';
import { HeadingSpring } from '../components/springs';
import {
  ImageSpring,
  ContentSpring,
  SocialMediaSpring,
} from '../components/springs/about';

const About = props => (
  <Layout>
    <Menu path={props.path} />
    <Spacer heightMobile={150} />
    <HeadingSpring>
      <MainHeading style={{ paddingLeft: '25px' }}>About</MainHeading>
    </HeadingSpring>
    <Flex style={{ paddingTop: '30px' }}>
      <ContentSpring>
        <AboutContent />
      </ContentSpring>
      <ImageSpring>
        <AboutImage />
      </ImageSpring>
    </Flex>
    <SocialMediaSpring>
      <Subheading style={{ paddingLeft: '25px' }}>Find me on...</Subheading>
      <SocialMedia />
    </SocialMediaSpring>
  </Layout>
);

export default About;
