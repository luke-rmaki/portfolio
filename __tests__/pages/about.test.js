import React from 'react';
import renderer from 'react-test-renderer';

import about from '../../src/pages/index';

describe('about', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<about />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
