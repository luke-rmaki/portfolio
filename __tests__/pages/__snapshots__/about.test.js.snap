// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about renders correctly 1`] = `<about />`;
